twinotter/external/goes/download_matching.py

```diff
@@ -0,0 +1,74 @@
+"""Download all GOES satellite images during the given flight.
+
+Data is downloaded to the current working directory
+
+Usage:
+    download_matching.py  <flight_data_path>
+    download_matching.py  (-h | --help)
+
+Arguments:
+    <flight_data_path>  Input flight data
+
+Options:
+    -h --help        Show help
+
+"""
+import pytz
+
+import worldview_dl
+
+from .. import goes
+from ... import load_flight, summary, util
+from ...util import scripting
+
+
+def main():
+    scripting.parse_docopt_arguments(get_images, __doc__)
+    return
+
+
+def get_images(
+        flight_data_path,
+        layer=goes.default_layer,
+        image_resolution=goes.default_spatial_resolution,
+        bbox=goes.default_bbox):
+    """Download all GOES images during the time of the specified flight
+
+    Args:
+        flight_data_path: The path to the MASIN flight data file or directory
+        layer: The GOES image layer to use. Also used as the prefix of the filename
+        image_resolution (float):
+        bbox: The spatial area to use in the format [S W N E] with units degrees
+    """
+    dataset = load_flight(flight_data_path)
+
+    date = summary.extract_date(dataset)
+    date = date.astimezone(pytz.utc)
+
+    start = date + summary.extract_time(dataset, 'time_coverage_start')
+    end = date + summary.extract_time(dataset, 'time_coverage_end')
+
+    start = util.round_datetime(start, goes.time_resolution)
```

Comment:
I found noticed recently that xarray has a [round](http://xarray.pydata.org/en/stable/generated/xarray.core.accessor_dt.DatetimeAccessor.round.html#xarray.core.accessor_dt.DatetimeAccessor.round) function for when you have `DataArray`s with datetime dtype. That might not be useful here though so I think you're accessing things from the legs CSV file, right?

Reply:
I have another pull request going at the moment (https://github.com/EUREC4A-UK/twin-otter/pull/12) where I've updated the use of the times. It's not too unified though. It's a mix between xarray, pandas and python datetime types. It would make sense to use the xarray times as much as possible but I've found they are painfully unintuitive a lot of the time as well. Anyway, not something too important here but I'll have a look into it.